versioned_docs/version-2.0/tutorials/backtesting.mdx

```diff
@@ -0,0 +1,185 @@
+    """Building a RAG Pipeline with LangChain and Amazon Bedrock Amazon Bedrock has great models for building LLM apps. This guide covers how to get started with them to build a RAG pipeline. https://gettingstarted.ai/langchain-bedrock/""",
+    """SF Meetup on March 27! Join our meetup to hear from LangChain and Pulumi experts and learn about building AI-enabled capabilities. Sign up: https://meetup.com/san-francisco-pulumi-user-group/events/299491923/?utm_campaign=FY2024Q3_Meetup_PUG%20SF&utm_content=286236214&utm_medium=social&utm_source=twitter&hss_channel=tw-837770064870817792""",
+    """Chat model response metadata @LangChainAI chat model invocations now include metadata like logprobs directly in the output. Upgrade your version of `langchain-core` to try it. PY: https://python.langchain.com/docs/modules/model_io/chat/logprobs JS: https://js.langchain.com/docs/integrations/chat/openai#generation-metadata""",
+    """Benchmarking Query Analysis in High Cardinality Situations Handling high-cardinality categorical values can be challenging. This blog explores 6 different approaches you can take in these situations. Blog: https://blog.langchain.dev/high-cardinality""",
+    """Building Google's Dramatron with LangGraph.js & Claude 3 We just released a long YouTube video (1.5 hours!) on building Dramatron using LangGraphJS and @AnthropicAI's Claude 3 "Haiku" model. It's a perfect fit for LangGraph.js and Haiku's speed. Check out the tutorial: https://youtube.com/watch?v=alHnQjyn7hg""",
+    """Document Loading Webinar with @AirbyteHQ Join a webinar on document loading with PyAirbyte and LangChain on 3/14 at 10am PDT. Features our founding engineer @eyfriis and the @aaronsteers and Bindi Pankhudi team. Register: https://airbyte.com/session/airbyte-monthly-ai-demo""",
+]
+
+_ = system.batch(
+    [{"messages": [HumanMessage(content=content)]} for content in inputs],
+    {"max_concurrency": 3},
+)
+```
+
+## Convert Prod Runs to Test
+
+The first step is to generate a dataset based on the production _inputs_.
+Then copy over all the traces to serve as a baseline run.
+
+```python
+from datetime import datetime, timedelta, timezone
+
+from langsmith import Client
+from langsmith.beta import convert_runs_to_test
+
+# How we are sampling runs to include in our test
+end_time = datetime.now(tz=timezone.utc)
+start_time = end_time - timedelta(days=1)
+run_filter = f'and(gt(start_time, "{start_time.isoformat()}"), lt(end_time, "{end_time.isoformat()}"))'
+
+
+# Fetch the runs we want to convert to a test
+client = Client()
+
+prod_runs = list(
+    client.list_runs(
+        project_name=project_name,
+        execution_order=1,
+        filter=run_filter,
+    )
+)
+
+# Name of the dataset we want to create
+dataset_name = f'{project_name}-backtesting {start_time.strftime("%Y-%m-%d")}-{end_time.strftime("%Y-%m-%d")}'
+# This converts the runs to a dataset + test
```

Comment:
By test are you referring to an experiment? If so we should change all of these to say "experiment" instead - we don't call them test projects anymore